doc/rtd/topics/datasources/ec2.rst

```diff
@@ -80,6 +81,13 @@ The settings that may be configured are:
  * **timeout**: the timeout value provided to urlopen for each individual http
    request.  This is used both when selecting a metadata_url and when crawling
    the metadata service. (default: 50)
+ * **apply_network_conig**: Boolean (default: True) to allow cloud-init
+   to configure any secondary NICs and secondoary IPs described by the
+   metadata service. Each network interface will enable dhcp4 to obtain
+   primary address and route. Secondary IPv4/IPv6 addresses are configured as
+   static IP addresses on the NIC if either ``local-ipv4s`` or ``ipv6s``
+   contain multiple addresses in the list. Each address greater than one
+   represented in that metadata is a secondary address.
```

Comment:
All network interfaces are configured with DHCP (v4) to obtain an primary IPv4 address and route. Interfaces which have a non-empty 'ipv6s' list will also enable DHCPv6 to obtain a primary IPv6 address and route.  The DHCP response (v4 and v6) return an IP that matches the first element of ``local-ipv4s`` and ``ipv6s`` lists respectively.   All additional values (secondary addresses) in the static ip lists will be added to interface.

Reply:
thanks for this